tests/unit/IPC/hipIpcEventHandle.cpp

```diff
@@ -0,0 +1,201 @@
+/*
+Copyright (c) 2015 - present Advanced Micro Devices, Inc. All rights reserved.
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+
+/* HIT_START
+ * BUILD: %t %s ../test_common.cpp NVCC_OPTIONS -std=c++11
+ * TEST: %t
+ * HIT_END
+ */
+
+#include "test_common.h"
+#include <sys/mman.h>
+#include <unistd.h>
+#include <fcntl.h>
+#include <sys/stat.h>
+#include <semaphore.h>
+
+#define N 10000
+#define Nbytes N*sizeof(int)
+
+int enablePeers(int dev0, int dev1) {
+    int canAccessPeer01, canAccessPeer10;
+    HIPCHECK(hipDeviceCanAccessPeer(&canAccessPeer01, dev0, dev1));
+    HIPCHECK(hipDeviceCanAccessPeer(&canAccessPeer10, dev1, dev0));
+    if (!canAccessPeer01 || !canAccessPeer10) {
+        return -1;
+    }
+
+    HIPCHECK(hipSetDevice(dev0));
+    HIPCHECK(hipDeviceEnablePeerAccess(dev1, 0 /*flags*/));
+    HIPCHECK(hipSetDevice(dev1));
+    HIPCHECK(hipDeviceEnablePeerAccess(dev0, 0 /*flags*/));
+    HIPCHECK(hipSetDevice(dev0));
+
+    return 0;
+};
+
+typedef struct mem_handle {
+    hipIpcEventHandle_t eventHandle;
+    int devCount;
+} hip_ipc_t;
+
+bool runipcTest()
+{
+    sem_t *sem_ob1, *sem_ob2;
+    
+    if ((sem_ob1 = sem_open ("/my-sem-object1", O_CREAT|O_EXCL, 0660, 0)) == SEM_FAILED) {
+        perror ("my-sem-object1");
+	exit (1);
+    }
+    if ((sem_ob2 = sem_open ("/my-sem-object2", O_CREAT|O_EXCL, 0660, 0)) == SEM_FAILED) {
+        perror ("my-sem-object2");
+	exit (1);
+    }
+
+    hip_ipc_t *shrd_mem=(hip_ipc_t *)mmap(NULL, sizeof(hip_ipc_t), PROT_READ | PROT_WRITE, MAP_SHARED | MAP_ANONYMOUS, 0, 0);
+    if (shrd_mem == MAP_FAILED) {
+        perror ("mmap failed");
+        exit (1);
+    }
+    
+    pid_t pid;
+
+    pid = fork();
+    
+    if(pid != 0) {
+    
+        int dev0 = 0;
+        int dev1 = 1;
```

Comment:
May I know why are we restricting the test only between devID =0 & devId=1? There can be a situation where even though setup is multi device but Dev0 and Dev1 do not have peer access but Dev0 and Dev3 has. I feel better approach would be to have a query function and prepare peer access list and then go for test.

Reply:
Modified the function to query for all the devices and find a combination of two devices with peer access on multi gpu setup.